Source/Core/Common/Config/Layer.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <list>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "Common/Config/Enums.h"
+#include "Common/Config/Section.h"
+
+namespace Config
+{
+using LayerMap = std::map<System, std::list<std::unique_ptr<Section>>>;
```

Comment:
That's a ton of pointer indirections? Is there anything that wouldn't work if this was unordered_map<..., vector<...>>?